File: src/VBNetFx/FAC_Generic/LayoutAndNumberformats.vb
Imports System.Drawing

Public Class LayoutAndNumberformats

    Private _U1Font As SerializableFontSetting
    Private _U2Font As SerializableFontSetting
    Private _U3Font As SerializableFontSetting
    Private _TableHeaderFont As SerializableFontSetting
    Private _TextAndTableBodyFont As SerializableFontSetting
    Private _SmallTableFont As SerializableFontSetting
    Private _HeaderFont As SerializableFontSetting
    Private _FooterFont As SerializableFontSetting
    'Private _LogoBitmap As Image
    Private _Gridstyle As FacessoLayoutGridstyle
    Private _HMinPrecision As Byte

    Sub New()
        _U1Font = New SerializableFontSetting("Arial", 22, FontStyle.Bold)
        _U2Font = New SerializableFontSetting("Arial", 16, FontStyle.Bold)
        _U3Font = New SerializableFontSetting("Arial", 13, FontStyle.Bold)
        _TableHeaderFont = New SerializableFontSetting("Arial", 10, FontStyle.Bold)
        _HeaderFont = New SerializableFontSetting("Arial", 10, FontStyle.Bold)
        _FooterFont = New SerializableFontSetting("Arial", 8, FontStyle.Regular)
        _TextAndTableBodyFont = New SerializableFontSetting("Arial", 9, FontStyle.Regular)
        _SmallTableFont = New SerializableFontSetting("Arial", 8, FontStyle.Regular)
        '_LogoBitmap = My.Resources.CompLogo_V2
        _Gridstyle = FacessoLayoutGridstyle.ThreeDGrid1
        _HMinPrecision = 2
    End Sub

    Public Property U1Font() As SerializableFontSetting
        Get
            Return _U1Font
        End Get
        Set(ByVal value As SerializableFontSetting)
            _U1Font = value
        End Set
    End Property

    Public Property U2Font() As SerializableFontSetting
        Get
            Return _U2Font
        End Get
        Set(ByVal value As SerializableFontSetting)
            _U2Font = value
        End Set
    End Property

    Public Property U3Font() As SerializableFontSetting
        Get
            Return _U3Font
        End Get
        Set(ByVal value As SerializableFontSetting)
            _U3Font = value
        End Set
    End Property

    Public Property TableHeaderFont() As SerializableFontSetting
        Get
            Return _TableHeaderFont
        End Get
        Set(ByVal value As SerializableFontSetting)
            _TableHeaderFont = value
        End Set
    End Property

    Public Property TextAndTableBodyFont() As SerializableFontSetting
        Get
            Return _TextAndTableBodyFont
        End Get
        Set(ByVal value As SerializableFontSetting)
            _TextAndTableBodyFont = value
        End Set
    End Property

    Public Property HeaderFont() As SerializableFontSetting
        Get
            Return _HeaderFont
        End Get
        Set(ByVal value As SerializableFontSetting)
            _HeaderFont = value
        End Set
    End Property

    Public Property FooterFont() As SerializableFontSetting
        Get
            Return _FooterFont
        End Get
        Set(ByVal value As SerializableFontSetting)
            _FooterFont = value
        End Set
    End Property

    Public Property SmallTableFont() As SerializableFontSetting
        Get
            Return _SmallTableFont
        End Get
        Set(ByVal value As SerializableFontSetting)
            _SmallTableFont = value
        End Set
    End Property

    'Public Property LogoBitmap() As Image
    '    Get
    '        Return _LogoBitmap
    '    End Get
    '    Set(ByVal value As Image)
    '        _LogoBitmap = value
    '    End Set
    'End Property

    Public Property Gridstyle() As FacessoLayoutGridstyle
        Get
            Return _Gridstyle
        End Get
        Set(ByVal value As FacessoLayoutGridstyle)
            _Gridstyle = value
        End Set
    End Property

    Public Property HMinPrecision() As Byte
        Get
            Return _HMinPrecision
        End Get
        Set(ByVal value As Byte)
            _HMinPrecision = value
        End Set
    End Property

    Public Function HminFormated(ByVal hmin As Double) As String
        Dim locFormat As String
        locFormat = "#,##0"
        If HMinPrecision > 0 Then
            locFormat &= "," & New String("0"c, HMinPrecision)
        End If
        Return hmin.ToString(locFormat)
    End Function

End Class

Public Enum FacessoLayoutGridstyle
    NoGrid
    SimpleGridThin
    SimpleGridThick
    ThreeDGrid1
    ThreeDGrid2
End Enum

Public Class SerializableFontSetting

    Private _FontName As String
    Private _FontSize As Single
    Private _FontStyle As FontStyle

    Sub New()
        FontName = "Arial"
        FontSize = 10
        FontStyle = FontStyle.Regular
    End Sub

    Sub New(ByVal font As Font)
        _FontName = font.Name
        _FontSize = font.Size
        _FontStyle = font.Style
    End Sub

    Sub New(ByVal fontName As String, ByVal fontSize As Single, ByVal fontStyle As FontStyle)
        _FontName = fontName
        _FontSize = fontSize
        _FontStyle = fontStyle
    End Sub

    Public Property FontName() As String
        Get
            Return _FontName
        End Get
        Set(ByVal value As String)
            _FontName = value
        End Set
    End Property

    Public Property FontSize() As Single
        Get
            Return _FontSize
        End Get
        Set(ByVal value As Single)
            _FontSize = value
        End Set
    End Property

    Public Property FontStyle() As FontStyle
        Get
            Return _FontStyle
        End Get
        Set(ByVal value As FontStyle)
            _FontStyle = value
        End Set
    End Property

    Public ReadOnly Property FontSettingsDescription() As String
        Get
            Return FontName & "; " & FontSize & " pt;" & FontStyle.ToString()
        End Get
    End Property

    Public Function ToFont() As Font
        Return New Font(FontName, FontSize, FontStyle)
    End Function
End Class

File: src/NetFxCSharp/Facesso.Generic/LayoutAndNumberformats.cs
using System.Drawing;

namespace Facesso
{
    public class LayoutAndNumberformats
    {
        private SerializableFontSetting _U1Font;
        private SerializableFontSetting _U2Font;
        private SerializableFontSetting _U3Font;
        private SerializableFontSetting _TableHeaderFont;
        private SerializableFontSetting _TextAndTableBodyFont;
        private SerializableFontSetting _SmallTableFont;
        private SerializableFontSetting _HeaderFont;
        private SerializableFontSetting _FooterFont;
        private FacessoLayoutGridstyle _Gridstyle;
        private byte _HMinPrecision;

        public LayoutAndNumberformats()
        {
            _U1Font = new SerializableFontSetting("Arial", 22, FontStyle.Bold);
            _U2Font = new SerializableFontSetting("Arial", 16, FontStyle.Bold);
            _U3Font = new SerializableFontSetting("Arial", 13, FontStyle.Bold);
            _TableHeaderFont = new SerializableFontSetting("Arial", 10, FontStyle.Bold);
            _HeaderFont = new SerializableFontSetting("Arial", 10, FontStyle.Bold);
            _FooterFont = new SerializableFontSetting("Arial", 8, FontStyle.Regular);
            _TextAndTableBodyFont = new SerializableFontSetting("Arial", 9, FontStyle.Regular);
            _SmallTableFont = new SerializableFontSetting("Arial", 8, FontStyle.Regular);
            _Gridstyle = FacessoLayoutGridstyle.ThreeDGrid1;
            _HMinPrecision = 2;
        }

        public SerializableFontSetting U1Font { get { return _U1Font; } set { _U1Font = value; } }
        public SerializableFontSetting U2Font { get { return _U2Font; } set { _U2Font = value; } }
        public SerializableFontSetting U3Font { get { return _U3Font; } set { _U3Font = value; } }
        public SerializableFontSetting TableHeaderFont { get { return _TableHeaderFont; } set { _TableHeaderFont = value; } }
        public SerializableFontSetting TextAndTableBodyFont { get { return _TextAndTableBodyFont; } set { _TextAndTableBodyFont = value; } }
        public SerializableFontSetting HeaderFont { get { return _HeaderFont; } set { _HeaderFont = value; } }
        public SerializableFontSetting FooterFont { get { return _FooterFont; } set { _FooterFont = value; } }
        public SerializableFontSetting SmallTableFont { get { return _SmallTableFont; } set { _SmallTableFont = value; } }
        public FacessoLayoutGridstyle Gridstyle { get { return _Gridstyle; } set { _Gridstyle = value; } }
        public byte HMinPrecision { get { return _HMinPrecision; } set { _HMinPrecision = value; } }

        public string HminFormated(double hmin)
        {
            string locFormat = "#,##0";
            if (HMinPrecision > 0)
                locFormat += "," + new string('0', HMinPrecision);
            return hmin.ToString(locFormat);
        }
    }

    public enum FacessoLayoutGridstyle
    {
        NoGrid,
        SimpleGridThin,
        SimpleGridThick,
        ThreeDGrid1,
        ThreeDGrid2
    }

    public class SerializableFontSetting
    {
        private string _FontName;
        private float _FontSize;
        private FontStyle _FontStyle;

        public SerializableFontSetting()
        {
            FontName = "Arial";
            FontSize = 10;
            FontStyle = FontStyle.Regular;
        }

        public SerializableFontSetting(Font font)
        {
            _FontName = font.Name;
            _FontSize = font.Size;
            _FontStyle = font.Style;
        }

        public SerializableFontSetting(string fontName, float fontSize, FontStyle fontStyle)
        {
            _FontName = fontName;
            _FontSize = fontSize;
            _FontStyle = fontStyle;
        }

        public string FontName { get { return _FontName; } set { _FontName = value; } }
        public float FontSize { get { return _FontSize; } set { _FontSize = value; } }
        public FontStyle FontStyle { get { return _FontStyle; } set { _FontStyle = value; } }

        public string FontSettingsDescription
        {
            get { return FontName + "; " + FontSize + " pt;" + FontStyle.ToString(); }
        }

        public Font ToFont()
        {
            return new Font(FontName, FontSize, FontStyle);
        }
    }
}
